zipkin/src/main/java/zipkin2/internal/Proto3Fields.java

```diff
@@ -190,15 +197,21 @@ static int decodeLowerHex(char c) {
 
     @Override String readValue(Buffer buffer, int length) {
       length *= 2;
-      char[] result = new char[length];
+
+      // All our hex fields are at most 32 characters.
+      if (length > 32) {
```

Comment:
maybe use assert here as that will eliminate the runtime check?

Reply:
Since the data comes over the wire, it's possible a bad client or something would send bad data. I think we would generally want a runtime check for incoming data which wouldn't be a local programming bug. What do you think?

Reply:
if that's possible, I'd throw an IllegalArgumentException as that's what we use for other malformed data at the moment.